webdriver/tests/bidi/network/remove_intercept/remove_intercept.py

```diff
@@ -39,12 +41,15 @@ async def test_remove_intercept(
 
     on_network_event = wait_for_event(f"network.{phase}")
 
-    # Request to top_context should be blocked and throw a ScriptEvaluateResultException
-    # from the AbortController.
-    with pytest.raises(ScriptEvaluateResultException):
-        await fetch(text_url)
+    # Request to top_context should be blocked.
+    # TODO(https://github.com/w3c/webdriver-bidi/issues/188): Use a timeout argument when available.
+    with pytest.raises(asyncio.TimeoutError):
+        await asyncio.wait_for(
+            asyncio.shield(bidi_session.browsing_context.navigate(context=top_context["context"], url=text_url, wait="complete")),
+            timeout=2.0,
+        )
```

Comment:
doesn't this add 2 sec for the test runtime?

Reply:
Yes.

Reply:
Instead, we can subscribe to some browsing-context specific event (?`browsingContext.load`?), and verify it was not emitted.

Reply:
Note that this PR is doing two different things: one is to introduce (and use) `wait_for_future_safe`, the other one is to rewrite the remove intercept test to depend on timeouts instead of exceptions (we discussed this in Matrix).

In case you have a concern with the latter but not with the former, I could split the latter into its own PR.

Reply:
> and verify it was not emitted.

Wouldn't this also depend on a timeout?

Reply:
Reverted. Splitting into another PR.